File: Patterns/code/java/DesignPatterns/src/main/java/templatemethod/TraditionalHouseBuilder.java
package templatemethod;

class TraditionalHouseBuilder extends HouseBuilder {

    @Override
    protected void buildWalls() {
        System.out.println("Building walls with bricks and mortar.");
    }

    @Override
    protected void installRoof() {
        System.out.println("Installing a pitched roof with clay tiles.");
    }

    @Override
    protected void doInterior() {
        System.out.println("Doing interior with a classic design and wood furniture.");
    }
}

File: code/csharp/DesignPatterns/templatemethod/TraditionalHouseBuilder.cs
using System;

namespace TemplateMethod
{
    class TraditionalHouseBuilder : HouseBuilder
    {
        protected override void BuildWalls()
        {
            Console.WriteLine("Building walls with bricks and mortar.");
        }

        protected override void InstallRoof()
        {
            Console.WriteLine("Installing a pitched roof with clay tiles.");
        }

        protected override void DoInterior()
        {
            Console.WriteLine("Doing interior with a classic design and wood furniture.");
        }
    }
}
